Tests/LibPDF/TestPDF.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2021, Simon Woertz <simon@woertz.at>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "AK/Forward.h"
+#include <AK/MappedFile.h>
+#include <AK/String.h>
+#include <LibPDF/Document.h>
+#include <LibTest/Macros.h>
+#include <LibTest/TestCase.h>
+
+TEST_CASE(linearized_pdf)
+{
+    auto file = MappedFile::map("/home/anon/Documents/pdf/linearized.pdf").release_value();
+    auto document = PDF::Document::create(file->bytes());
+    EXPECT_EQ(document->get_page_count(), 1U);
+}
+
+TEST_CASE(non_linearized_pdf)
+{
+    auto file = MappedFile::map("/home/anon/Documents/pdf/non-linearized.pdf").release_value();
+    auto document = PDF::Document::create(file->bytes());
+    EXPECT_EQ(document->get_page_count(), 1U);
+}
+
+TEST_CASE(complex_pdf)
+{
+    auto file = MappedFile::map("/home/anon/Documents/pdf/complex.pdf").release_value();
+    auto document = PDF::Document::create(file->bytes());
+    EXPECT_EQ(document->get_page_count(), 3U);
+}
+
+TEST_CASE(empty_file_issue_10702)
+{
+    AK::ReadonlyBytes empty;
+    auto document = PDF::Document::create(empty);
+    EXPECT(document.is_null());
+}
+
+TEST_CASE(truncated_pdf_issue_10717)
```

Comment:
Small nitpick. `LibPDF` uses consume/peek extensively. It is likely that there are many truncation errors.

This specific issue lies within `parse_header`. Suggest:

```suggestion
TEST_CASE(truncated_pdf_header_issue_10717)
```

Admittedly, this is not the only bug in `parse_header` due to truncation either.

Reply:
Done.